trieste/models/model_interfaces.py

```diff
@@ -499,6 +511,32 @@ def evaluate_loss_of_model_parameters() -> tf.Tensor:
         multiple_assign(self.model, current_best_parameters)
 
 
+class NumDataPropertyMixin:
+    """Mixin class for exposing num_data as a property, stored in a tf.Variable. This is to work
```

Comment:
What does Mixin mean?

Reply:
A mixin is "a class that contains methods for use by other classes without having to be the parent class of those other classes". It's useful here as both wrappers want the same behaviour.